File: javav2/example_code/route53/src/main/java/com/example/route/DeleteHostedZone.java
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.example.route;

// snippet-start:[route53.java2.delete_hosted_zone.main]
// snippet-start:[route53.java2.delete_hosted_zone.import]
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.route53.Route53Client;
import software.amazon.awssdk.services.route53.model.DeleteHostedZoneRequest;
import software.amazon.awssdk.services.route53.model.Route53Exception;
// snippet-end:[route53.java2.delete_hosted_zone.import]

/**
 * Before running this Java V2 code example, set up your development
 * environment, including your credentials.
 *
 * For more information, see the following documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class DeleteHostedZone {
    public static void main(String[] args) {
        final String usage = """

                Usage:
                    <hostedZoneId>\s

                Where:
                    hostedZoneId - The hosted zone id.\s
                """;

        if (args.length < 1) {
            System.out.println(usage);
            System.exit(1);
        }

        String hostedZoneId = args[0];
        Region region = Region.AWS_GLOBAL;
        Route53Client route53Client = Route53Client.builder()
                .region(region)
                .build();

        delHostedZone(route53Client, hostedZoneId);
        route53Client.close();
    }

    public static void delHostedZone(Route53Client route53Client, String hostedZoneId) {
        try {
            DeleteHostedZoneRequest deleteHostedZoneRequestRequest = DeleteHostedZoneRequest.builder()
                    .id(hostedZoneId)
                    .build();

            route53Client.deleteHostedZone(deleteHostedZoneRequestRequest);
            System.out.println("The hosted zone was deleted");

        } catch (Route53Exception e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
    }
}
// snippet-end:[route53.java2.delete_hosted_zone.main]

File: kotlin/services/route53/src/main/kotlin/com/kotlin/route/DeleteHostedZone.kt
// snippet-comment:[These are tags for the AWS doc team's sample catalog. Do not remove.]
// snippet-sourcedescription:[DeleteHostedZone.kt demonstrates how to delete a hosted zone.]
// snippet-keyword:[AWS SDK for Kotlin]
// snippet-service:[Amazon Route 53]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.route

// snippet-start:[route53.kotlin.delete_hosted_zone.import]
import aws.sdk.kotlin.services.route53.Route53Client
import aws.sdk.kotlin.services.route53.model.DeleteHostedZoneRequest
import kotlin.system.exitProcess
// snippet-end:[route53.kotlin.delete_hosted_zone.import]

/**
Before running this Kotlin code example, set up your development environment,
including your credentials.

For more information, see the following documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main(args: Array<String>) {

    val usage = """
    Usage:
        <hostedZoneId> 

    Where:
        hostedZoneId - The hosted zone id.
    """

    if (args.size != 1) {
        println(usage)
        exitProcess(0)
    }

    val hostedZoneId = args[0]
    delHostedZone(hostedZoneId)
}

// snippet-start:[route53.kotlin.delete_hosted_zone.main]
suspend fun delHostedZone(hostedZoneId: String?) {

    val deleteHostedZoneRequestRequest = DeleteHostedZoneRequest {
        id = hostedZoneId
    }

    Route53Client { region = "AWS_GLOBAL" }.use { route53Client ->
        route53Client.deleteHostedZone(deleteHostedZoneRequestRequest)
        println("The hosted zone with id $hostedZoneId was deleted")
    }
}
// snippet-end:[route53.kotlin.delete_hosted_zone.main]
